internal/store/postgres/resource/repository.go

```diff
@@ -53,27 +53,81 @@ func (r Repository) Update(ctx context.Context, resourceModel *resource.Resource
 	return nil
 }
 
+func (r Repository) Delete(ctx context.Context, resourceModel *resource.Resource) error {
+	res := FromResourceToModel(resourceModel)
+
+	deleteResourceQuery := `UPDATE resource SET status=$5, updated_at=NOW()
+                WHERE full_name=$1 AND store=$2 AND project_name = $3 And namespace_name = $4`
+	tag, err := r.db.Exec(ctx, deleteResourceQuery, res.FullName, res.Store, res.ProjectName, res.NamespaceName, resource.StatusDeleted)
+	if err != nil {
+		return errors.Wrap(resource.EntityResource, "error updating resource to database", err)
+	}
+
+	if tag.RowsAffected() == 0 {
+		return errors.NotFound(resource.EntityResource, "no resource to delete for "+res.FullName)
+	}
+	return nil
+}
+
 func (r Repository) ChangeNamespace(ctx context.Context, res *resource.Resource, newTenant tenant.Tenant) error {
+	existingResource, err := r.ReadByFullName(ctx, newTenant, res.Store(), res.FullName(), false)
+	if err != nil && !errors.IsErrorType(err, errors.ErrNotFound) {
+		return err
+	}
+
+	tx, err := r.db.Begin(ctx)
+	if err != nil {
+		return errors.Wrap(resource.EntityResource, "error begin db transaction", err)
+	}
+
+	err = r.hardDelete(ctx, tx, existingResource)
+	if err != nil {
+		_ = tx.Rollback(ctx)
+		return err
+	}
+
 	updateResource := `UPDATE resource SET namespace_name=$1, updated_at=now()
 	WHERE full_name=$2 AND store=$3 AND project_name = $4 And namespace_name = $5`
-	tag, err := r.db.Exec(ctx, updateResource,
+	tag, err := tx.Exec(ctx, updateResource,
 		newTenant.NamespaceName(), res.FullName(), res.Store(),
 		res.Tenant().ProjectName(), res.Tenant().NamespaceName())
 	if err != nil {
+		_ = tx.Rollback(ctx)
 		return errors.Wrap(resource.EntityResource, "error changing tenant for resource:"+res.FullName(), err)
 	}
-
 	if tag.RowsAffected() == 0 {
+		_ = tx.Rollback(ctx)
 		return errors.NotFound(resource.EntityResource, "no resource to changing tenant for ")
 	}
-	return nil
+
+	err = tx.Commit(ctx)
+	return errors.WrapIfErr(resource.EntityResource, "error commit db transaction", err)
```

Comment:
do we still return error if the commit is succeeded?

Reply:
i believe it will not sending any error on err is nil https://github.com/goto/optimus/blob/08a335984e7758244910cc806a6102844c8fde9e/internal/errors/errors.go#L171 should be handled by the function